src/main/java/org/broadinstitute/consent/http/db/mapper/UserWithRolesReducer.java

```diff
@@ -44,6 +51,19 @@ public void accumulate(Map<Integer, User> map, RowView rowView) {
     } catch(MappingException e) {
       //Ignore institution mapping errors, possible for new users to not have an institution
     }
+    //user role join can cause duplication of data if done in tandem with joins on other tables
+    //ex) The same LC can end up being repeated multiple times
+    //Below only adds LC if not currently saved on the array
+    try {
+      if(Objects.nonNull(rowView.getColumn("lc_id", Integer.class))) {
+        LibraryCard lc = rowView.getRow(LibraryCard.class);
+        if(!user.getLibraryCards().stream().anyMatch(card -> card.getId() == lc.getId())) {
+          user.getLibraryCards().add(lc);
+        }
```

Comment:
We started, at one point, to use the add to list pattern which obviates the need for initialization in lines 26-29. See the User addRole and addProperty methods for examples.